### PR TITLE
Build with MSVC

### LINKED ISSUE #27: Support MSVC

### PR DESCRIPTION
**Pull request by matthew-d-jones**
Closes #27 

- Builds with MSVC on Windows 10, mostly changes to CMake, just a couple of tweaks to the code to be cross-platform.
- Updated documentation to say it builds on MSVC
- Using `override` instead of `virtual` in a few places

### CODE REVIEW

**Comment by matthew-d-jones** [comment]
The two failing tests are due to a 10054 error ("connection reset by peer") being returned instead of `asio::error::misc_errors::eof` after the connection has been terminated. I _think_ this may simply be an expected difference between Linux and Windows. @SkyToGround do you know anything more about this?

**Comment by SkyToGround** [comment]
@matthew-d-jones I do not know anything more than you in this case but your suggestion seems likely to me. I suggest that you simply modify the code to accept both values.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,20 +1,42 @@
-cmake_minimum_required(VERSION 3.9)
+cmake_minimum_required(VERSION 3.7)
 project("graylog-logger"
-  VERSION 1.1.0
+  VERSION 1.2.0
```
[the remark this answers is not in the document]
Comment (matthew-d-jones) [reply CMakeLists.txt]: I will update the conan package as there'll likely be changes required to make the recipe work on Windows.

File: src/GraylogConnection.cpp [hunk 2]
```diff
@@ -13,6 +13,10 @@
 #include <ciso646>
 #include <utility>
 
+namespace {
```
[the remark this answers is not in the document]
Comment (matthew-d-jones) [reply src/GraylogConnection.cpp]: Correct. The solution I employed here is the same we are using the Forwarder and File Writer.

File: cmake/conan.cmake [hunk 3]
```diff
@@ -0,0 +1,532 @@
+# The MIT License (MIT)
```
[the remark this answers is not in the document]
Comment (matthew-d-jones) [reply cmake/conan.cmake]: We've been using it for a couple of years now in FW&FW and NeXus-Streamer, it is convenient not having to manually run conan. Simply use `-DCONAN=MANUAL` with CMake to revert to the behaviour you had before.

File: src/GraylogConnection.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (SkyToGround) [reply src/GraylogConnection.cpp]: Ok. Looking forward to being able to use C++17 (and thus `[[maybe_unused]]`) on this project in the near (ha!) future then.
LGTM.

**Pull request merged**